storages/backends/gcloud.py

```diff
@@ -0,0 +1,233 @@
+from tempfile import SpooledTemporaryFile
+
+from django.core.exceptions import ImproperlyConfigured
+from django.core.files.base import File
+from django.core.files.storage import Storage
+from django.utils.deconstruct import deconstructible
+from django.utils.encoding import force_bytes, force_text, smart_str
+from django.utils import timezone
+from storages.utils import clean_name, safe_join, setting
+
+try:
+    from google.cloud.storage.client import Client
+    from google.cloud.storage.blob import Blob
+    from google.cloud.exceptions import NotFound
+except ImportError:
+    raise ImproperlyConfigured("Could not load Google Cloud Storage bindings.\n"
+                               "See https://github.com/GoogleCloudPlatform/gcloud-python")
+
+
+class GoogleCloudFile(File):
+    def __init__(self, name, mode, storage):
+        self.name = name
+        self._mode = mode
+        self._storage = storage
+        self.blob = storage.bucket.get_blob(name)
+        if not self.blob and 'w' in mode:
+            self.blob = Blob(self.name, storage.bucket)
+        self._file = None
+        self._is_dirty = False
+
+    @property
+    def size(self):
+        return self.blob.size
+
+    def _get_file(self):
+        if self._file is None:
+            self._file = SpooledTemporaryFile(
+                max_size=self._storage.max_memory_size,
+                suffix=".GSStorageFile",
+                dir=setting("FILE_UPLOAD_TEMP_DIR", None)
+            )
+            if 'r' in self._mode:
+                self._is_dirty = False
+                self.blob.download_to_file(self._file)
+                self._file.seek(0)
+        return self._file
+
+    def _set_file(self, value):
+        self._file = value
+
+    file = property(_get_file, _set_file)
+
+    def read(self, num_bytes=None):
+        if 'r' not in self._mode:
+            raise AttributeError("File was not opened in read mode.")
+        if num_bytes is not None:
+            return super(GoogleCloudFile, self).read(num_bytes)
+        else:
+            return super(GoogleCloudFile, self).read()
```

Comment:
Unfortunately we do. [Django defines the default argument to be `None`](https://docs.djangoproject.com/en/dev/ref/files/file/#django.core.files.File.read) but `SpooledTemporaryFile` (and Python `file` objects in general) can't handle the case where the first argument (`size`) is `None`:

```
Traceback (most recent call last):
  File "/home/ubuntu/gcs_integration/westore/tests.py", line 31, in test_write_to_storage
    self.assertEqual(f.read(), 'storage contents')
  File "/home/ubuntu/gcs_integration/storages/backends/gcloud.py", line 56, in read
    return super(GoogleCloudFile, self).read(num_bytes)
  File "/usr/lib/python2.7/tempfile.py", line 585, in read
    return self._file.read(*args)
TypeError: an integer is required
```

Instead the argument has to be omitted or set to a negative number.

The reason you haven't run into this in the `s3boto` backend is because [it uses `*args and **kwargs`](https://github.com/jschneier/django-storages/blob/e52a127523fdd5be50bb670ccad566c5d527f3d1/storages/backends/s3boto.py#L134-L137), which means that if the user omits the argument there's no `None` to pass through to the underlying `read()`. I believe if the user _does_ specify `num_bytes=None` s3boto will break, but I haven't tested this.

I agree that `num_bytes` is an unfortunate name 😿.